tests/test_layers.py

```diff
@@ -89,16 +92,34 @@ def test_retrieve_adm_area_names_level_0(self, mocker):
         adm_area = AdmArea(country="Timor-Leste", level=0)
         assert adm_area.retrieve_adm_area_names() == ["Timor-Leste"]
 
-    def test_retrieve_adm_area_names_level_1(self, mocker):
-        data = {
-            'id': [0, 1],
-            'COUNTRY': ["Mock Country", "Mock Country"],
-            'NAME_1': ["Mock Region 1", "Mock Region 2"],
-            'geometry': [None, None],
-        }
-        mock_gdf = gpd.GeoDataFrame(data, crs='EPSG:4326')
-
+    def test_retrieve_adm_area_names_level_1(self, mocker, mock_gdf):
         mocker.patch("gpbp.layers.GADMDownloader.get_shape_data_by_country_name", return_value=mock_gdf)
         adm_area = AdmArea(country="Timor-Leste", level=1)
 
         assert np.array_equal(adm_area.retrieve_adm_area_names(), np.array(["Mock Region 1", "Mock Region 2"]))
+
+
+class TestAdmAreaGetAdmArea:
+    def test_get_adm_area_level_0(self, mocker):
+        mocker.patch("gpbp.layers.AdmArea._get_country_data")
+        adm_area = AdmArea(country="Timor-Leste", level=0)
+        adm_area.get_adm_area("Any Region")
+
+        assert adm_area.geometry is None
```

Comment:
This is actually not the behavior that we want: we want geometry to be the country’s geometry, and adm_name to be the name of the country. I suggest we xfail this one

Reply:
That's a great idea! I've never xfailed anything, clearly I never truly worked in a test-driven-development way ;)

Reply:
Alternatively, we use the patch for get_shape_data_by_country_name like you did in the next tests, but I don’t like that we’re going _into_ another method. Let’s discuss

Reply:
I'm actually not sure how to properly do the xfail. I have tried the following:

```
    @pytest.mark.xfail(
        reason="The geometry should always be set regardless of the level, but it is not in this case.",
        strict=True
    )
    def test_get_adm_area_level_0(self, mocker, mock_level0_gdf):
        mocker.patch("gpbp.layers.GADMDownloader.get_shape_data_by_country_name", return_value=mock_level0_gdf)
        adm_area = AdmArea(country="Timor-Leste", level=0)
        adm_area.get_adm_area("Any Region")

        assert adm_area.geometry == mock_level0_gdf.geometry[0]
```

However this test actually passes, because, during the init, since this is a level 0 administrative area, the geometry will be set 🤦 

How did you envision this test? I'm not sure what to do here 🥲

Reply:
As I see it now, what we ultimately would like is for this (and only this) function to set the name and geometry of an administrative area, regardless of the level. With that in mind, I’d design the test almost as you had it:

```

    @pytest.mark.xfail(reason="adm_name and geometry not set for level 0. Refactor")
    def test_get_adm_area_level_0(self, mocker):
        mocker.patch("gpbp.layers.AdmArea._get_country_data”) 
        adm_area = AdmArea(country="Timor-Leste", level=0)
        adm_area.get_adm_area("Any Region”)
        
        assert adm_area.adm_name == "Timor-Leste"
        assert adm_area.geometry is not None
```

Does that make sense to you? 

And how do you add smileys to these comments?

Reply:
I see what you mean, yes that's a good alternative. I have updated it :) Thanks for the review 🙏 

to add emojis either you write `:<name of the emoji>:` or you use the emoji key on the mac keyboard (for me it's the `fn` key)

Reply:
:smile: